azafea/event_processors/endless/metrics/v3/model/__init__.py

```diff
@@ -570,6 +570,64 @@ def _get_fields_from_payload(payload: GLib.Variant) -> Dict[str, Any]:
         }
 
 
+class SplitFlatpakRepoStats(SingularEvent):
+    """Statistics for splitting the Flatpak repo from the OSTree repo.
+
+    Prior to Endless OS 4, both the OS and Flatpaks were stored in a
+    single, shared OSTree repository. On the first boot into Endless OS
+    4 or later, these are split into two repositories, one for the OS
+    and one for Flatpaks. This can take a long time, and blocks the boot
+    process, so we report this event to quantify how long it takes and
+    why.
+
+    This is a one-time operation which only runs if needed, so this
+    event is sent at most once per Endless OS system.
+
+    :UUID name: ``SPLIT_FLATPAK_REPO_STATS`` in eos-boot-helper
+
+    .. versionadded:: 4.0.1
+    """
+    __tablename__ = 'split_flatpak_repo_stats_v3'
+    __event_uuid__ = '880fd091-2e68-4bd2-baa1-f6810fabcc1c'
+    __payload_type__ = '(uuuuuuutut)'
+
+    #: time elapsed in seconds
+    elapsed = Column(Integer, nullable=False)
```

Comment:
If you were feeling fancy you could store this as an `interval` instead. [sqlalchemy](https://docs.sqlalchemy.org/en/14/core/type_basics.html#sqlalchemy.types.Interval) maps this to `datetime.timedelta` and an `Interval` column type.

This would be nice because it allows the `psql` command line (and Metabase) to format the duration nicely:

```
azafea=> create temporary table blahblah (number int, int interval);
CREATE TABLE
azafea=> insert into blahblah values (86400, '86400 seconds');
INSERT 0 1
azafea=> select * from blahblah;
 number |   int    
--------+----------
  86400 | 24:00:00
(1 row)
```

Reply:
OK, I'll take a look.